src/NumericalIntegration.jl

```diff
@@ -147,8 +150,9 @@ end
 function integrate(x::AbstractVector, y::AbstractVector, m::RombergEven)
     @assert length(x) == length(y) "x and y vectors must be of the same length!"
     @assert ((length(x) - 1) & (length(x) - 2)) == 0 "Need length of vector to be 2^n + 1"
-    maxsteps = Int(log2(length(x)-1))
-    rombaux = zeros(eltype(y), maxsteps, 2)
+    maxsteps::Integer = Int(log2(length(x)-1))
+    T = _zero_type(x,y)
+    rombaux = zeros(T, maxsteps, 2)
```

Comment:
I guess you don't have to add the heuristic `T = _zero_type(x, y)` here if you just compute the initial entry `(y[1] + y[end]) * h * HALF` first and then create the array. I.e., something like
```suggestion
    @inbounds v = (y[1] + y[end]) * h * HALF
    rombaux = Matrix{typeof(v)}(undef, maxsteps, 2)
    @inbounds rombaux[1, 1] = v
```

Reply:
you would need to have the computation of `h` happen before line 154 currently it is in line 158

Reply:
Sure, the computations would have to be rearranged and the duplicate lines be removed.

Reply:
I do prefer this method by the way. I've used this a few times when I need to initialize an array to an arbitrary type.

Reply:
I like both ways - the `T = ...` method seems more prescriptive whereas having an explicit first element `v` seems more flexible. Which one did you prefer @m-wells ? I'm not sure which method you meant by "this method".

Reply:
I prefer the calculation of the first element and then calling `typeof` on the result. I also reuse the result and skip the first element in the subsequent loop. It is type-stable and it is less error prone than doing the type inference "directly".

Reply:
Done!